client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-item.js

```diff
@@ -64,53 +61,55 @@ class LabelItem extends Component {
 
 		const { orderId, siteId, translate } = this.props;
 
-		const openDialog = e => {
-			e.preventDefault();
+		const openDialog = () => {
 			this.props.openReprintDialog( orderId, siteId, label.labelId );
 		};
 
 		return (
-			<span>
-				<ReprintDialog siteId={ siteId } orderId={ orderId } { ...label } />
-				<Button onClick={ openDialog } borderless className="shipping-label__button">
-					{ translate( 'Reprint' ) }
-				</Button>
-			</span>
+			<PopoverMenuItem onClick={ openDialog } icon="print">
+				{ translate( 'Reprint' ) }
+			</PopoverMenuItem>
 		);
 	};
 
 	renderLabelDetails = label => {
 		const { orderId, siteId, translate } = this.props;
 
-		const openDialog = e => {
-			e.preventDefault();
+		const openDialog = () => {
 			this.props.openDetailsDialog( orderId, siteId, label.labelId );
 		};
 
 		return (
-			<span>
-				<DetailsDialog siteId={ siteId } orderId={ orderId } { ...label } />
-				<Button onClick={ openDialog } borderless className="shipping-label__button">
-					{ translate( 'View details' ) }
-				</Button>
-			</span>
+			<PopoverMenuItem onClick={ openDialog } icon="info-outline">
+				{ translate( 'View details' ) }
+			</PopoverMenuItem>
 		);
 	};
 
 	render() {
-		const { label, translate } = this.props;
+		const { siteId, orderId, label, translate } = this.props;
 
 		return (
 			<div key={ label.labelId } className="shipping-label__item">
 				<p className="shipping-label__item-detail">
-					<span>
-						{ translate( 'Label #%(labelIndex)s printed', {
-							args: {
-								labelIndex: label.labelIndex + 1,
-							},
-						} ) }
-					</span>
-					{ label.showDetails && this.renderLabelDetails( label ) }
+					{ translate( '%(service)s label (#%(labelIndex)d) printed', {
+						args: {
+							service: label.serviceName,
+							labelIndex: label.labelIndex + 1,
+						},
+					} ) }
+
+					<DetailsDialog siteId={ siteId } orderId={ orderId } { ...label } />
+					<RefundDialog siteId={ siteId } orderId={ orderId } { ...label } />
+					<ReprintDialog siteId={ siteId } orderId={ orderId } { ...label } />
```

Comment:
Should these get moved into the conditional below? The old code only rendered these components if `label.showDetails`.

Reply:
Thanks – fixed in https://github.com/Automattic/wp-calypso/pull/25393/commits/c511c4f572cd46104f3b7816980fa0fa78deb70b by wrapping the menu and dialogs into a span, and verified no effect on layout.